File: Contest/Biweekly/Biweekly Contest 169/3737. Count Subarrays With Majority Element I/Solution.java
class Solution {
    public int countMajoritySubarrays(int[] nums, int target) {
        // N : len(nums)
        // Time Complexity : O(N^2)
        // Space Complexity : O(1)
        int n = nums.length;
        int subCount = 0;
        for(int i = 0; i < n; i++) {
            int targetCount = 0;
            for (int j = i; j < n; j++) {
                if (target == nums[j]) targetCount += 1;
                if (targetCount > (j - i + 1) / 2) subCount += 1;
            }
        }
        return subCount;
    }
}

File: Contest/Biweekly/Biweekly Contest 169/3737. Count Subarrays With Majority Element I/Solution.py
from typing import List

class Solution:
    def countMajoritySubarrays(self, nums: List[int], target: int) -> int:
        # N : len(nums)
        # Time Complexity : O(N^2)
        # Space Complexity : O(1)
        n = len(nums)
        subCount = 0
        for i in range(n):
            countTarget = 0
            for j in range(i, n):
                if nums[j] == target:
                    countTarget += 1
                if countTarget > (j - i + 1) // 2:
                    subCount += 1
        return subCount